### PR TITLE
Add verify_pyi script to validate .pyi stubs in distributions

### PR DESCRIPTION
**Pull request by masenf**
### Type of change

- [x] New feature (non-breaking change which adds functionality)

### Description

Adds a new `scripts/verify_pyi.py` script that validates built distributions (wheels and sdists) contain the `.pyi` type stub files they are expected to generate. This script is run by the publish workflow's post-build hook (`.github/scripts/publish/post_build.sh`) to prevent releasing packages without type information.

**Why this matters:** Reflex generates type stubs at build time (via `scripts/hatch_build.py` for the root package and the `hatch-reflex-pyi` hook for component packages). Since `*.pyi` files are gitignored, only the build process creates them. A build that silently fails to generate stubs would ship a release with no type information, which is not recoverable since PyPI versions are immutable.

**How it works:**
- Reads each package's `pyproject.toml` to determine if it generates stubs (by checking for the `reflex-pyi` hook or `*.pyi` artifact declarations)
- For packages that should generate stubs, verifies every built artifact (wheel and sdist) contains at least one `.pyi` file
- Fails the build if any artifact is missing stubs, preventing the release from proceeding to the approval gate

**Changes:**
- `scripts/verify_pyi.py`: New script with functions to read build configuration, detect stub-generating packages, count stubs in artifacts, and validate distributions
- `tests/units/test_verify_pyi.py`: Comprehensive unit tests covering all code paths, including edge cases (empty dist dirs, mixed build matrices, non-distribution files)
- `.github/scripts/publish/post_build.sh`: Updated to call the new verification script instead of the previous reflex-only check
- `pyproject.toml`: Added linting exception for the new script's use of `print()`

### Testing

- [x] Added 233 lines of comprehensive unit tests in `tests/units/test_verify_pyi.py`
- [x] Tests cover: configuration parsing, stub detection logic, artifact inspection, error cases, and real package detection
- [x] Tests validate both synthetic test cases and real packages in the repository (reflex and component packages)
- [x] All tests pass with the implementation

### Checklist

- [x] Tests pass with adequate coverage
- [x] Code follows project conventions (Google-style docstrings, concise implementation)
- [x] Linting configured appropriately for the new script

https://claude.ai/code/session_01NwKbDTae52hrg5HY63whHb

<a href="https://cubic.dev/pr/reflex-dev/reflex/pull/6961?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

### CODE REVIEW

File: scripts/verify_pyi.py [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+import tarfile
+import zipfile
+from collections.abc import Mapping
+from pathlib import Path
+from typing import Any
+
+if sys.version_info >= (3, 11):
+    import tomllib
+else:
+    import tomli as tomllib  # pyright: ignore[reportMissingImports]
```
[the remark this answers is not in the document]
Comment (masenf) [reply scripts/verify_pyi.py:42]: Valid — reproduced it. Running the verifier the old way on 3.10 gave exactly this:

```
$ UV_PYTHON=3.10 ... uv run --no-project python scripts/verify_pyi.py
ModuleNotFoundError: No module named 'tomli'
```

Fixed in 9980b981 by declaring the backport as inline script metadata and running the hook with `uv run --script`, so the dependency comes from the script itself rather than from an environment it never had:

```python
# /// script
# requires-python = ">=3.10"
# dependencies = [
#     "tomli; python_version < '3.11'",
# ]
# ///
```

Same block `scripts/check_min_deps.py` carries, for the same reason. `--no-config` keeps the ephemeral resolution clear of the workspace's uv settings.

Verified against a real `uv build` on 3.10 (installs the backport, passes), on 3.14 and with no `UV_PYTHON` set, across the pass, skip and stub-stripped-wheel paths. Worth noting the publish job itself runs 3.14, where `tomllib` is stdlib and the block resolves to no dependencies at all — so the release path installs nothing new either way.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Review by FarhanAliRaza — APPROVED** [review]

**Comment by masenf** [comment]
thanks for the fixup
